pkg/render/common/clusterrole/clusterrolerules.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2020-2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterrole
+
+import (
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	"github.com/tigera/operator/pkg/render/common/securitycontextconstraints"
+)
+
+// CalicoKubeControllersClusterRoleRules defines RBAC rules for calico-kube-controller pods
+// and for guardian to handle kube-controller requests from the management cluster.
+func CalicoKubeControllersClusterRoleRules(isEnterpriseCluster bool, isOpenShift bool, isManagedCluster bool) []rbacv1.PolicyRule {
+	rules := KubeControllersRoleCommonRules(isOpenShift)
+	if isEnterpriseCluster {
+		rules = append(rules, KubeControllersRoleEnterpriseCommonRules(isManagedCluster)...)
+		rules = append(rules,
+			rbacv1.PolicyRule{
```

Comment:
It would be nice to see commenting explaining why each set of permissions is needed (like we do for other roles)